framework/SupervisedLearning/DynamicModeDecomposition.py

```diff
@@ -31,35 +28,99 @@
 
 #Internal Modules------------------------------------------------------------------------------------
 from utils import mathUtils
-from SupervisedLearning import supervisedLearning
+from utils import InputData, InputTypes
+from SupervisedLearning import SupervisedLearning
 #Internal Modules End--------------------------------------------------------------------------------
 
 
-class DynamicModeDecomposition(supervisedLearning):
+class DMD(SupervisedLearning):
   """
     This surrogate is aimed to construct a "time-dep" surrogate based on
     Dynamic Mode Decomposition method.
     Ref. Kutz, Brunton, Brunton, Proctor. Dynamic Mode Decomposition:
         Data-Driven Modeling of Complex Systems. SIAM Other Titles in
         Applied Mathematics, 2016
   """
-  def __init__(self, **kwargs):
+  info = {'problemtype':'regression', 'normalize':True}
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    specs = super().getInputSpecification()
+    specs.description = r"""The \xmlString{DMD} ROM aimed to construct a time-dependent (or any other monotonic
+        variable) surrogate model based on Dynamic Mode Decomposition
+        This surrogate is aimed to perform a ``dimensionality reduction regression'', where, given time
+        series (or any monotonic-dependent variable) of data, a set of modes each of which is associated
+        with a fixed oscillation frequency and decay/growth rate is computed
+        in order to represent the data-set.
+        In order to use this Reduced Order Model, the \xmlNode{ROM} attribute
+        \xmlAttr{subType} needs to be set equal to \xmlString{DMD}.
+        \\
+        Once the ROM  is trained (\textbf{Step} \xmlNode{RomTrainer}), its parameters/coefficients can be exported into an XML file
+        via an \xmlNode{OutStream} of type \xmlAttr{Print}. The following variable/parameters can be exported (i.e. \xmlNode{what} node
+        in \xmlNode{OutStream} of type \xmlAttr{Print}):
+        \begin{itemize}
+          \item \xmlNode{rankSVD}, see XML input specifications above
```

Comment:
We still need this, since these are the exported parameters and coefficients of the ROM, and we need the full list of them. The generated input will only contains the input parameters.